### PR TITLE
migrate to github actions

### PR DESCRIPTION
**Pull request by jjergus**
- bumping HHVM and HHAST version to ^4.80 (it might work with 4.59 or 4.60 but 4.80 is the first version that works and is still supported)
- do we want MacOS here?
- there are 2 extra steps here besides the shared action, verified that they can fail with output as expected:
  - https://github.com/jjergus/hack-codegen/runs/1745345674
  - https://github.com/jjergus/hack-codegen/runs/1745353796

### CODE REVIEW

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+          - latest
+          - nightly
+    runs-on: ${{matrix.os}}-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: hhvm/actions/hack-lint-test@master
+        with:
+          hhvm: ${{matrix.hhvm}}
+      # Check the codegen hasn't been modified; this is also done as part of the
+      # next step, but we want to make sure the verify-signatures executable works
```
Comment (fredemmott) [inline .github/workflows/build-and-test.yml:26]: we should probably end with an intentional + inverted failure :)

**Review by fredemmott — APPROVED** [review]
MacOS: this is a dev tool so ideally, but the odds of it breaking in a platform-specific manner seem so low it's not worth the increased time to results

File: .github/workflows/build-and-test.yml (hunk 1 above)
Comment (jjergus) [reply .github/workflows/build-and-test.yml:26]: > this is a dev tool

Which of our projects is not? :D

> we should probably end with an intentional + inverted failure :)

What do you mean, should I swap the order of those 2 steps?

Comment (fredemmott) [reply .github/workflows/build-and-test.yml:26]: Add a new step:

```
echo >> some/codegen/file
bin/verify-codegen
```

... but expect failure then

Comment (jjergus) [reply .github/workflows/build-and-test.yml:26]: https://github.com/jjergus/hack-codegen/runs/1751696414

**Pull request merged**